src/Polly.Core/Registry/RegistryPipelineComponentBuilder.cs

```diff
@@ -30,23 +30,25 @@ internal class RegistryPipelineComponentBuilder<TBuilder, TKey>
         _configure = configure;
     }
 
-    internal PipelineComponent CreateComponent()
+    internal (TBuilder builder, PipelineComponent component) CreateComponent()
```

Comment:
```suggestion
    internal PipelineComponent CreateComponent()
```

Can we get away without returning returning the builder instance?

Reply:
we need the builder instance somehow to flow the the pool from builder -> pipeline -- considering I'm somewhat new here ( 😄 ), I wanted to be as minimally invasive as possible.

Some options I'm thinking are:

1. Hold the builder instance on a `Builder` property (i initially had this and it got a bit messy / invasive so i unwound it. 
 its also a bit confusing behavior when you factor in the reload functionality), happy to do it though.
2. Refactor component builder into a `ResiliencePipelineFactory` and pass the `DisposeBehavior` into the factory.

It would look something like:

```csharp
public class ResiliencePipelineFactory<TBuilder, TKey> // pass in DisposeBehavior into the ctor
{ 
    public ResilientPipeline CreatePipeline()
}
```

I'm in favor of option 2 as it provides a better delineation of responsibility between registering/state managing the pipelines and the actual creation.  Right now the registry does a little of both.

Reply:
Let's go with option 1 with this small change:

`internal (ResilienceContextPool pool, PipelineComponent component) CreateComponent()`

Returning full builder is an overkill as we just need the pool instance.

If you are interested in pursuing options #2 you can create a follow-up draft PR where we can discuss the change. If it makes code simpler I am all for it.

Reply:
Otherwise the changes look good :)

Reply:
@martintmk I went with what I had instead of option 1, but reduced the return to just the contextPool as requested.

Adding the `Builder` property onto the component builder could potentially have some weird side effects in a later code state that read from this property due to reload behavior and would prefer to keep the builder instance transient.

Reply:
After this PR is merged, I can contribute a proposal PR that does the light refactoring for option 2, which will improve the overall health of this sub-component.